### PR TITLE
Issue #4: Added the ability to add new items to the list and save to the db

### PR DESCRIPTION
**Pull request by aidantorrence**
## Description

Created a form for users to enter item names and how soon they anticipate purchasing items (soon, kind of soon, not soon).  When users submit the form, it is saved to db associated with user's token (we generated a token 'plushy cuny idiom').  When submitting an item, a number corresponding to estimated number of days is saved.  Item names are displayed as an unordered list when added.  Deleted the button from issue #1 which was no longer needed

AC:

- [ ] User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
- [ ] When the user submits the form, the item is saved to the database, associated with the user’s token
- [ ] Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
- [ ] The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)
- [ ] Item names should be displayed as a simple, unordered list on the list view after they’re added

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before

image

image

### After

image

image

image

### CODE REVIEW

File: src/List.js [hunk 1]
```diff
@@ -1,11 +1,41 @@
...
+  onSnapshot,
+  doc,
+  setDoc,
+} from '@firebase/firestore';
+import { db } from './lib/firebase.js';
 
 export function List() {
+  const [items, setItems] = useState([]);
+
+  useEffect(() => {
```
Comment (gabynr) [inline src/List.js:15]: In the Component App you copied and modified according to the issue the Useefect Hook From App.js to List.js here  and moved the method handleClick from App.js to the new AddForm, why didn't you delete this code on the App component?

File: src/List.js [hunk 2]
```diff
@@ -1,11 +1,41 @@
...
+    const fetchItems = async () => {
+      const response = collection(db, 'shopping-list');
+      const itemList = query(response);
+
+      const unsubscribe = onSnapshot(itemList, (querySnapshot) => {
+        const items = querySnapshot.docs.reduce((acc, doc) => {
+          const { name } = doc.data();
+          const id = doc.id;
+          return [...acc, { id, name }];
```
Comment (gabynr) [inline src/List.js:24]: I have a question, I think that in the UseEffect you are not searching by token, which I believe means that you'll get all the items in the database. Can you test creating an item for a different list with another token, if you see all the items that are in the database?

**Review by gabynr — CHANGES_REQUESTED** [review]
Great Work!!! Your code is clear, I can read it without problem. I love that. I have a few questions 👏🏽

File: src/List.js (hunk 2 above)
Comment (gabynr) [reply src/List.js:24]: Yes, maybe it isn't you scope in this issue . Thanks for doing that test for me!!

**Comment by andreamartz** [comment]
@aidantorrence @bajancode 

@gabynr was approving on behalf of both of us. We probably could have been more clear about that, so my apologies. I just realized this morning that you are still waiting for my review.

Now that @gabynr and I have merged our PR (a few days ago), there are merge conflicts with your PR that need to be addressed. It should be pretty straightforward. Please let me know if it's not, and I can help.

**Pull request merged**